packages/common/aws.js

```diff
@@ -610,22 +615,62 @@ class S3ListObjectsV2Queue {
 }
 exports.S3ListObjectsV2Queue = S3ListObjectsV2Queue;
 
-exports.checksumS3Objects = (algorithm, bucket, key, options = {}) => {
-  const param = { Bucket: bucket, Key: key };
+/**
+ * Calculate checksum for S3 Object
+ *
+ * @param {Object} params - params
+ * @param {string} params.algorithm - checksum algorithm
+ * @param {string} params.bucket - S3 bucket
+ * @param {string} params.key - S3 key
+ * @param {Object} [params.options] - crypto.createHash options
+ *
+ * @returns {number|string} - calculated checksum
+ */
+exports.calculateS3ObjectChecksum = async ({
+  algorithm,
+  bucket,
+  key,
+  options
+}) => {
+  const fileStream = exports.getS3ObjectReadStream(bucket, key);
+  return generateChecksumFromStream(algorithm, fileStream, options);
+};
 
-  if (algorithm.toLowerCase() === 'cksum') {
-    return getFileChecksumFromStream(
-      exports.s3().getObject(param).createReadStream()
-    );
+/**
+ * Validate S3 object checksum against expected sum
+ *
+ * @param {Object} params - params
+ * @param {string} params.algorithm - checksum algorithm
+ * @param {string} params.bucket - S3 bucket
+ * @param {string} params.key - S3 key
+ * @param {number|string} params.expectedSum - expected checksum
+ * @param {Object} [params.options] - crypto.createHash options
+ *
+ * @throws {InvalidChecksum} - Throws error if validation fails
+ * @returns {boolean} - returns true for success
+ */
+exports.validateS3ObjectChecksum = async ({
+  algorithm,
+  bucket,
+  key,
+  expectedSum,
+  options
+}) => {
+  const fileStream = exports.getS3ObjectReadStream(bucket, key);
+  if (await validateChecksumFromStream(algorithm, fileStream, expectedSum, options)) {
+    return true;
   }
+  const msg = `Invalid checksum for S3 object s3://${bucket}/${key} with type ${algorithm} and expected sum ${expectedSum}`;
+  throw new errors.InvalidChecksum(msg);
+};
 
-  return new Promise((resolve, reject) => {
-    const hash = crypto.createHash(algorithm, options);
-    const fileStream = exports.s3().getObject(param).createReadStream();
-    fileStream.on('error', reject);
-    fileStream.on('data', (chunk) => hash.update(chunk));
-    fileStream.on('end', () => resolve(hash.digest('hex')));
-  });
+// Maintained for backwards compatibility
+exports.checksumS3Objects = (algorithm, bucket, key, options = {}) => {
```

Comment:
Aside: it's weird that this was named `checksumS3Objects` when it was only operating on one object

Reply:
Part of the reason I deprecated it.